doc/rest-authentication.md

```diff
@@ -0,0 +1,106 @@
+# Glusterd2 REST API authentication
+
+Glusterd2 REST API authentication is based on [JWT](https://jwt.io). If REST
+authentication is enabled then each client request should include
+`Authorization` header as example below.
+
+    Authorization: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnbHVzdGVyY2xpIiwiaWF0IjoxNTMxNzI5NzQyLCJleHAiOjE1MzE3Mjk3NTJ9._WIwO7PrHUHIT62SdzfkyNjqD1GEgX2cYqN8ACZCtaw
+
+**Note**: REST Authentication can be disabled by adding `restauth=false`
+in Glusterd2 config file(Default path is
+`/var/lib/glusterd2/glusterd2.toml` in case of rpm installation)
+
+## Default user
+
+When `glusterd2` starts for the first time, it creates the
+`$GLUSTERD2_STATEDIR/auth` file which will contain the secret. If
+"gluster" user group is available in the system then this `auth` file
+can be read by any user in that machine who are part of "gluster"
+group.
+
+**Note**: If "gluster" user group is not available during first start
+of `glusterd2` it limits the read permission to `root:root`
+
+## glustercli
+
+With default installation, `glustercli` will know the location of
+`auth` file. `glustercli` will generate JWT token using the secret
+available in `auth` file and attach it with every REST API calls.
+
+Default installation will not require any change to use `glustercli`.
+
+If `auth` file is in different path(When `glusterd2` is running with
+custom `workdir`), then run `glustercli` by specifying
+`--secret-file`. For example,
+
+    glustercli --secret-file=/root/setup1/glusterd2/auth peer status
+
+**Note**: bash alias can be added like `alias glustercli='glustercli
+--secret-file=/root/setup1/glusterd2/auth`
+
+Secret is taken by `glustercli` in following order of precedence
+(highest to lowest):
+
+    --secret
+    --secret-file
+    GLUSTERD2_AUTH_SECRET (environment variable)
+    --secret-file (default path)
+
+## Curl example
```

Comment:
Should be under ReST API section?

Reply:
A link should be added from the ReST API document to this doc.